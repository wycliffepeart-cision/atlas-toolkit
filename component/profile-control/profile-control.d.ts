export declare class ProfileControl extends HTMLElement {
}
//# sourceMappingURL=profile-control.d.ts.map