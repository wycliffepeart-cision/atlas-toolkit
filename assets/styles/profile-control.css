@layer atlas {
	atlas-profile-control {
		--avatar-as-button-hover-filter: drop-shadow(var(--atlas-shadow-soft-100))
			drop-shadow(var(--atlas-shadow-hard-100));
	}

	atlas-profile-control {
		--text-name-text-font-size: 1.25em;
		--text-name-font-weigth: 700;
		--text-name-margin-left: 1em;
		--container-spacing: 0em;
		--icon-margin-left: 0.6rem;

		display: inline-grid;
		grid-template-columns: max-content;
		grid-template-areas: "img icon name" "menu menu menu";
		justify-content: stretch;
		align-items: stretch;
	}

	atlas-profile-control::part(image) {
		grid-area: img;
	}

	atlas-profile-control atlas-avatar:hover {
		filter: var(--avatar-as-button-hover-filter);
	}

	atlas-profile-control > [slot="icon"] {
		grid-area: icon;
		margin-left: var(--icon-margin-left);
		aspect-ratio: 1/1;
		width: 0.7em;
	}

	atlas-profile-control::part(name) {
		height: 100%;
		grid-area: name;
		flex-wrap: nowrap;
		margin-inline-start: var(--text-name-margin-left);
		font-size: var(--text-name-text-font-size);
		font-weight: var(--text-name-font-weigth);
	}

	atlas-profile-control::part(menu) {
		display: block;
		grid-area: menu;
	}

	atlas-profile-control::part(name),
	atlas-profile-contro > [slot="icon"] {
		display: flex;
		height: 100%;
		align-items: center;
	}
}
