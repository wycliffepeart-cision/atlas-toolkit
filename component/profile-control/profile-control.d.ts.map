{"version":3,"file":"profile-control.d.ts","sourceRoot":"","sources":["../../../src/component/profile-control/profile-control.ts"],"names":[],"mappings":"AAGA,qBACa,cAAe,SAAQ,WAAW;CAAG"}