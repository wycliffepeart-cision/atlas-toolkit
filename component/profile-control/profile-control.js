import { _ as __decorate } from '../../tslib.es6-2e0713a3.js';
import { Template } from '../../decorators/template.js';

var ProfileControlTemplate = "<template>\n  <slot name=\"image\" part=\"image\"></slot>\n  <slot name=\"icon\" part=\"icon\"></slot>\n  <slot name=\"name\" part=\"name\"></slot>\n  <slot name=\"menu\" part=\"menu\"></slot>\n</template>\n";

let ProfileControl = class ProfileControl extends HTMLElement {
};
ProfileControl = __decorate([
    Template(ProfileControlTemplate)
], ProfileControl);
if (!customElements.get('atlas-profile-control'))
    customElements.define('atlas-profile-control', ProfileControl);

export { ProfileControl };
