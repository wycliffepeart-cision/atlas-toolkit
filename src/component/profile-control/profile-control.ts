import { Template } from '../../decorators/template';
import ProfileControlTemplate from './profile-control.html';

@Template(ProfileControlTemplate)
export class ProfileControl extends HTMLElement {}

if (!customElements.get('atlas-profile-control')) customElements.define('atlas-profile-control', ProfileControl);