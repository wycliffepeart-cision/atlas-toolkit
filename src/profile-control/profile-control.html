<template>
	<slot name="image" part="image"></slot>
	<slot name="icon" part="icon"></slot>
	<slot name="name" part="name"></slot>
	<slot name="menu" part="menu"></slot>
</template>
