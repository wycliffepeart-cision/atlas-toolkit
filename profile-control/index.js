import { ProfileControl } from './profile-control.js';
import '../tslib.es6-806de115.js';
import '../decorators/template.js';

if (!customElements.get('atlas-profile-control'))
    customElements.define('atlas-profile-control', ProfileControl);
