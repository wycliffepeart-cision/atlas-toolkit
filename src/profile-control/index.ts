import { ProfileControl } from './profile-control';

if (!customElements.get('atlas-profile-control')) customElements.define('atlas-profile-control', ProfileControl);
