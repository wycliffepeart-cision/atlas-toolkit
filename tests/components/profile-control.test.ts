import '../../src/component/profile-control';

const TAG = 'atlas-profile-control';
const getShadowDomContent = (element) => element.shadowRoot.innerHTML;

describe('Profile Control', () => {
  it('is correctly appended in the DOM', async () => {
    await customElements.whenDefined(TAG);
    const element = document.createElement(TAG);
    document.body.appendChild(element);

    (expect(element) as any).toBeInTheDocument();
  });

  it('is capable of being selected', async () => {
    await customElements.whenDefined(TAG);
    const element = document.createElement(TAG);
    document.body.appendChild(element);

    const selectedElement = document.querySelector(TAG);
    (expect(selectedElement) as any).toBeInTheDocument();
  });

  it('has the correct content when created', async () => {
    await customElements.whenDefined(TAG);
    const element = document.createElement(TAG);
    document.body.appendChild(element);

    (expect(element) as any).toContainHTML('<atlas-profile-control></atlas-profile-control>');
    (expect(element) as any).toBeEmptyDOMElement();
    (expect(getShadowDomContent(element)) as any).toMatchSnapshot();
  });
});
