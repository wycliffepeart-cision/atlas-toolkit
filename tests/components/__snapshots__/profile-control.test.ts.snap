// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Control has the correct content when created 1`] = `
<slot name="image"
      part="image"
>
</slot>
<slot name="icon"
      part="icon"
>
</slot>
<slot name="name"
      part="name"
>
</slot>
<slot name="menu"
      part="menu"
>
</slot>
`;
